### ISSUE TITLE
Add Regressor

### ISSUE THREAD

**Issue by mitmul** [chainer/chainer#5964]
I got some requests from users on adding `Regressor` class besides `Classifier`.

`Classifier` can actually be used to calculate regression losses like MSE by specifying `mean_squared_error` etc. to the `lossfun` argument of `Classifier`, but using `Classifier` for regression tasks seems confusing in the code base. And `accfun` of `Classifier` should also be changed for an appropriate one.

So some users want to use `Regressor` class as a separate class from `Classifier`.

**Comment by mitmul** [comment]
Update: From the offline discussion, we may need to design a new way to provide such a class for loss calculation and reporting. The current recommended way can be seen as to write a model as a loss function, but it may be better to separate the model itself from loss calculation parts. That would also be useful for another potential feature like inference abstraction, so now it's time to re-design how to write/use a loss function.

**Comment by crcrpar** [comment]
This shouldn’t be closed, IMO.

**Comment by fiarabbit** [comment]
@crcrpar I agree with your opinion, because of the recent post in SO(https://stackoverflow.com/questions/56111935/regression-with-chainer).

**Comment by corochann** [comment]
JFYI: chainer chemistry supports `Regressor` class which additionally supports `predict` method.
https://github.com/pfnet-research/chainer-chemistry/blob/master/chainer_chemistry/models/prediction/regressor.py

**Comment by Saanidhyavats** [comment]
I want to contribute on this issue, can anyone guide me on this issue?

**Comment by Saanidhyavats** [comment]
Hi, I want to contribute on this issue, how shall I start?

**Comment by ankahira** [comment]
This is not resolved yet.

**Comment by mitmul** [comment]
I think it's difficult to add a new feature to Chainer because it has shifted to maintenance phase (ref. https://chainer.org/announcement/2019/12/05/released-v7.html).
OTOH, PyTorch already provides a lot of loss functions including MSELoss. https://pytorch.org/docs/stable/nn.html#loss-functions
